runtime/compiler/control/CompilationThread.cpp

```diff
@@ -5408,6 +5425,24 @@ TR::CompilationInfo::getNextMethodToBeCompiled(TR::CompilationInfoPerThread *com
             nextMethodToBeCompiled = getJProfilingCompQueue().extractFirstCompRequest();
             }
          }
+#if defined(J9VM_OPT_JITSERVER)
+      else if (getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT &&
```

Comment:
These tests need to take into consideration the fact that the test above`getLowPriorityCompQueue().hasLowPriorityRequest() &&
canProcessLowPriorityRequest())` can respond `true` and we may schedule the compilation too soon 9while server is still not available)